lib/iris/fileformats/um/_fast_load_structured_fields.py

```diff
@@ -88,15 +88,11 @@ def data(self):
         if not self._structure_calculated:
             self._calculate_structure()
         if self._data_cache is None:
-            data_arrays = [da.from_array(f._data, f._data.shape)
-                           for f in self.fields]
-            vector_dims_list = list(self.vector_dims_shape)
-            vector_dims_list.reverse()
-            self._data_cache = data_arrays
-            for size in vector_dims_list:
-                self._data_cache = [da.stack(self._data_cache[i:i+size]) for i
-                                    in range(0, len(self._data_cache), size)]
-            self._data_cache, = self._data_cache
+            data_arrays = np.array([f._data for f in self.fields],
```

Comment:
I think you need to convert each `f._data` into a dask array here. This isn't done by `multidim_daskstack`, unless calling `da.stack` converts the inputs into dask arrays and then stacks them? Either way, the conversion to arrays is not being done by the common API from #2421, which it probably should be.